File: backend/src/main/java/org/xxg/backend/backend/dto/ApiResponse.java
package org.xxg.backend.backend.dto;

import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

/**
 * 统一API响应包装类
 * <p>所有接口返回值的统一格式，包含状态、消息和数据三部分</p>
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ApiResponse<T> {
    /** 请求是否成功 */
    private boolean success;
    /** 响应消息 */
    private String message;
    /** 响应数据 */
    private T data;

    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, "success", data);
    }
    public static <T> ApiResponse<T> ok(String message, T data) {
        return new ApiResponse<>(true, message, data);
    }
    public static <T> ApiResponse<T> ok(String message) {
        return new ApiResponse<>(true, message, null);
    }
    public static <T> ApiResponse<T> error(String message) {
        return new ApiResponse<>(false, message, null);
    }
}

File: mobile/shared/src/commonMain/kotlin/com/xxgkami/shared/model/ApiResponse.kt
package com.xxgkami.shared.model

import kotlinx.serialization.Serializable

/**
 * 通用API响应包装类
 * 所有API接口统一返回此格式，通过泛型 [T] 支持不同数据类型
 *
 * @param T 响应数据的类型
 * @property success 请求是否成功
 * @property message 响应消息（通常用于错误提示）
 * @property data 响应数据，可能为null
 */
@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val message: String? = null,
    val data: T? = null
)
